LDKNodeMonday/View Model/Home/Receive/ReceiveViewModel.swift

```diff
@@ -0,0 +1,328 @@
+                await MainActor.run {
+                    self.receiveViewError = .init(
+                        title: "Address error",
+                        detail: "Failed to generate any addresses."
+                    )
+                }
+            }
+        }
+    }
+
+    func maxReceiveCapacity() -> UInt64 {
+        var maxReceiveCapacity = UInt64(0)
+        let channels = lightningClient.listChannels()
+        for channel in channels {
+            if channel.inboundCapacityMsat > maxReceiveCapacity {
+                maxReceiveCapacity = channel.inboundCapacityMsat
+            }
+        }
+        return maxReceiveCapacity
+    }
+
+    func parseUnifiedQR(_ unifiedQR: String) -> [PaymentAddress?] {
+        // Split the string by '?'
+        let components = unifiedQR.components(separatedBy: "?")
+
+        guard components.count > 1 else { return [] }
+
+        // Extract onchain (everything before the first '?') and remove the "BITCOIN:" prefix
+        var onchain = components[0]
+        if onchain.lowercased().hasPrefix("bitcoin:") {
+            onchain = String(onchain.dropFirst(8))  // Remove "BITCOIN:"
+        }
+
+        // Join the rest of the components back together
+        let remainingString = components.dropFirst().joined(separator: "?")
+
+        // Split the remaining string by '&'
+        let params = remainingString.components(separatedBy: "&")
+
+        var bolt11: String?
+        var bolt12: String?
+
+        for param in params {
+            if param.lowercased().starts(with: "lightning=lnbc") {
+                bolt11 = String(param.dropFirst("lightning=".count))
+            } else if param.lowercased().starts(with: "lightning=lntb") {
+                bolt11 = String(param.dropFirst("lightning=".count))
+            } else if param.lowercased().starts(with: "lightning=lno") {
+                bolt12 = String(param.dropFirst("lightning=".count))
+            } else if param.lowercased().starts(with: "lno=") {
+                bolt12 = String(param.dropFirst("lno=".count))
+            }
+        }
+
+        let unifiedQRString = unifiedQRString(
+            onchainAddress: onchain,
+            amountBTC: amountSat.satsAsBTC,
+            message: message,
+            bolt11: bolt11,
+            bolt12: bolt12
+        )
+
+        let paymentAddresses: [PaymentAddress?] = [
+            PaymentAddress(
+                type: .bip21,
+                address: unifiedQRString
+            ),
+            PaymentAddress(
+                type: .onchain,
+                address: onchain
+            ),
+            bolt11.map {
+                PaymentAddress(
+                    type: .bolt11,
+                    address: $0
+                )
+            },
+            bolt12.map {
+                PaymentAddress(
+                    type: .bolt12,
+                    address: $0
+                )
+            },
+        ]
+
+        return paymentAddresses
+    }
+}
```

Comment:
thought for a follow up PR or something, but I'd like to stuff this in `BitcoinUI` because it'd be a nice spot for people like myself to grab reusable/tested parsing like this